src/modules/strings.wren

```diff
@@ -1,24 +1,28 @@
+import "ensure" for Ensure
+
 #module=essentials
 class Strings {
     static upcase(s) {
-        return s.bytes.map { |x|
-            if ((LOWERCASE_A..LOWERCASE_Z).contains(x)) x = x - 32
-            return String.fromByte(x)
-        }.join("")
+        Ensure.string(s, "s")
+        upcase_(s)
     }
+    foreign static upcase_(s)
 
     static downcase(s) {
-        return s.bytes.map { |x|
-            if ((UPPERCASE_A..UPPERCASE_Z).contains(x)) x = x + 32
-            return String.fromByte(x)
-        }.join("")
+        Ensure.string(s, "s")
+        downcase_(s)
     }
+    foreign static downcase_(s)
+
     static capitalize(s) {
+        Ensure.string(s, "s")
         if (s.isEmpty) return ""
         if (s.count == 1) return Strings.upcase(s)
         return Strings.upcase(s[0]) + s[1..-1]
     }
+
     static titlecase(s) {
+        Ensure.string(s, "s")
```

Comment:
Just now saw this... I'm not sure we need extra validation EVERYWHERE (vs just C dispatch)... if a bad type is passed the VM can throw a regular error to inform the user... I'm not sure ALL our APIs need this just for "niceness"... it's worth looking around and seeing what the precedent is on this.

I honestly don't mind if someone passes a number to a string function and gets a weird error - then has to sort it out.  I think this type of "over-achiever" type checking for system libraries just slows things down.